### PR TITLE
Add tracking for other vehicles.

### PR DESCRIPTION
**Pull request by alvkao58**
Adds functionality for perfect tracking of other vehicles in the environment. Locations of vehicles are calculated and logged with respect to the ego vehicle.

### CODE REVIEW

File: configs/data_gatherer.conf [hunk 1]
```diff
@@ -18,8 +18,10 @@
 --carla_camera_image_height=600
 #--nocarla_random_player_start
 #--carla_start_player_num=1
+--perfect_tracking
+--perfect_tracking_num_steps=10
```
Comment (ICGog) [inline configs/data_gatherer.conf]: Can you please also update data_gatherer_autopilot.conf?

File: pylot/simulation/perfect_tracker_operator.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            while (len(buffer) > 0 and buffer[0].timestamp < timestamp):
+                buffer.popleft()
+            if len(buffer) == 0:
+                return False
+            assert buffer[0].timestamp == timestamp
+        return True
+
+    def on_notification(self, msg):
+        # Pop the oldest message from each buffer.
+        with self._lock:
```
Comment (ICGog) [inline pylot/simulation/perfect_tracker_operator.py]: We'll fix ERDOS so that we don't have to do this :)

File: pylot.py [hunk 3]
```diff
@@ -137,6 +137,12 @@ def add_detection_component(graph, bgr_camera_setup, camera_ops, carla_op):
                           [fusion_op])
             graph.connect([fusion_op, carla_op], [fusion_verif_op])
 
+    perfect_tracker_ops = []
```
Comment (ICGog) [inline pylot.py]: Can you please rename perfect_tracker_ops to tracker_ops, and also store the tracker_op in the list? Thus, we'll return the real object tracker if --obj_tracking flag is enabled, or the perfect tracker if the --perfect_tracking flag is enabled.

Comment (alvkao58) [reply pylot.py]: Added comment indicating why we're keeping them separate for now.

File: configs/data_gatherer.conf (hunk 1 above)
Comment (alvkao58) [reply configs/data_gatherer.conf]: Done.

**Pull request merged**